view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java

```diff
@@ -368,28 +374,106 @@ public static boolean sendPinotTableCreationRequest(
     return sendPinotTableCreationRequest(
         pinotTableSpec.getControllerHost(),
         pinotTableSpec.getControllerPort(),
-        tableConfig.toJsonString());
+        tableConfig.toJsonString(),
+        tableConfig.getTableName());
   }
 
   public static boolean sendPinotTableCreationRequest(
-      String controllerHost, String controllerPort, final String tableConfig) {
+      String controllerHost,
+      String controllerPort,
+      final String tableConfig,
+      final String tableName) {
+    String controllerAddress = getControllerAddressForTableCreate(controllerHost, controllerPort);
+    LOGGER.info("Trying to send table creation request {} to {}. ", tableConfig, controllerAddress);
+    int responseCode = sendRequest("POST", controllerAddress, tableConfig);
+    if (responseCode == 409) {
+      LOGGER.info("Trying to update table with request {} to {}. ", tableConfig, controllerAddress);
+      sendRequest(
+          "PUT",
+          getControllerAddressForTableUpdate(controllerHost, controllerPort, tableName),
+          tableConfig);
+    }
+    return true;
+  }
+
+  /**
+   * @param requestMethod: Type of request (PUT/GET/POST/DELTE)
+   * @param urlString: Api endpoint
+   * @param payload: Payload
+   * @return Status Code
+   */
+  public static int sendRequest(String requestMethod, String urlString, String payload) {
+    HttpURLConnection conn = null;
+    try {
+      final URL url = new URL(urlString);
+
+      conn = (HttpURLConnection) url.openConnection();
+      conn.setDoOutput(true);
+      conn.setRequestMethod(requestMethod);
+
+      final BufferedWriter writer =
+          new BufferedWriter(
+              new OutputStreamWriter(conn.getOutputStream(), StandardCharsets.UTF_8));
+      writer.write(payload, 0, payload.length());
+      writer.flush();
+
+      final int responseCode = conn.getResponseCode();
+      if (responseCode != 200) {
```

Comment:
static variable for the response code 200, also use the same while returning

Reply:
Okay